docs/index.rst

```diff
@@ -158,6 +158,30 @@ pointing to where in the given object validation failed, and a human readable
      }
    }
 
+Read-only fields
+~~~~~~~~~~~~~~~~
+Many of the API's :ref:`data structures <data>` contain read-only fields provided in API responses, for example, API-generated ``id`` fields, or a resource's ``url`` field. If read only fields are provided by clients in API request bodies, the way these are handled depends on the request method.
```

Comment:
I would maybe write `return read-only fields in API responses` rather than `contain read-only fields provided in API responses`.